go/state/gostate/archive_state.go

```diff
@@ -155,14 +155,21 @@ func (s *ArchiveState) Export(ctx context.Context, out io.Writer) (common.Hash,
 		return common.Hash{}, state.ExportNotSupported
 	}
 
-	exportableTrie := mptio.NewExportableArchiveTrie(trie, s.block)
-	rootHash, err := mptio.ExportLive(ctx, mptio.NewLog(), exportableTrie, out)
+	err := mptio.ExportBlockFromOnlineArchive(ctx, trie, out, s.block)
+	if err != nil {
+		// export opens the trie parallel to the current program,
+		// possible error is sent to the caller but should not corrupt
+		// the database on error.
```

Comment:
It might ... if the error happened during the initial flush ... would this be recorded by the flush?

Reply:
good point. It seems it was not checked, I also found a few places where I think error should be stored as well but was not. I will add it to the other PR to simplify this one

Reply:
I created a new PR #1007 for this. I think this check was missing in Flush. when it is added I think we can keep the code here as described in the comment